### PR TITLE
Fill water level gaps

### PR DESCRIPTION
**Pull request by amsnyder**
Fill internal water level data gaps using linear interpolation, as discussed in issue #58 . Leading and trialing nans are ignored when applying butterworth filter. Inputs and outputs are the same as before, but with no missing data

### CODE REVIEW

**Comment by amsnyder** [comment]
I can push the data to S3 once we approve this pull request

File: 02_munge/src/munge_noaa_nos.py [hunk 1]
```diff
@@ -30,6 +30,29 @@ def read_data(raw_datafile, read_location, s3_bucket):
...
+    #early exit if there are no nans  
+    if not bd.any():
+        return x
+
+    #find nonnans index numbers
+    gd = np.flatnonzero(~bd)
+
+    #ignore leading and trailing nans
+    bd[:gd.min()]=False
+    bd[(gd.max()+1):]=False
```
Comment (galengorski) [inline 02_munge/src/munge_noaa_nos.py:49]: This is just the first and last entry correct?

**Review by galengorski — APPROVED** [review]
This will output at hourly time step correct? Just want to make sure, otherwise looks good to me

File: 02_munge/src/munge_noaa_nos.py (hunk 1 above)
Comment (amsnyder) [reply 02_munge/src/munge_noaa_nos.py:49]: yes, still hourly! it is the same as whatever time step goes in

**Pull request merged**